src/services/shims.ts

```diff
@@ -779,6 +798,35 @@ module ts {
                 });
         }
 
+        public getTSConfigFileInfo(fileName: string, sourceTextSnapshot: IScriptSnapshot): string {
+            return this.forwardJSONCall(
+                "getTSConfigFileInfo('" + fileName + "')",
+                () => {
+                    let text = sourceTextSnapshot.getText(0, sourceTextSnapshot.getLength());
+
+                    try {
+                        var json = /\S/.test(text) ? JSON.parse(text) : {};
+                    }
+                    catch (e) {
+                        return {
+                            options: {},
+                            files: [],
+                            errors: realizeDiagnostic(createCompilerDiagnostic(Diagnostics.Unable_to_open_file_0, fileName), '\r\n')

```

Comment:
ts.ReadConfigFile reads the configfile from disk, since we already have the contents of the tsconfig.json file we don't need to read the file at this point. Further this would incur additional cost when communicating with the host.